src/apps/search-result/useFilterHandler.tsx

```diff
@@ -1,47 +1,43 @@
-import { useState } from "react";
-import { FilterItemTerm, TermOnClickHandler } from "./types";
-
-export type Filter = {
-  [key: string]: { [key: FilterItemTerm["key"]]: FilterItemTerm };
-};
+import { useCallback } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  add,
+  remove,
+  clear,
+  FilterPayloadType,
+  Filter
+} from "../../core/filter.slice";
+import { RootState } from "../../core/store";
+import {
+  removeUrlQueryParam,
+  setQueryParametersInUrl
+} from "../../core/utils/helpers/url";
 
 const useFilterHandler = () => {
-  const [filters, setFilters] = useState<Filter>({});
+  const dispatch = useDispatch();
+  const filters = useSelector((state: RootState) => state.filter) as Filter;
+
+  const clearFilter = useCallback(() => {
+    removeUrlQueryParam("filters");
+    dispatch(clear());
+  }, [dispatch]);
 
-  const filterHandler: TermOnClickHandler = ({
-    filterItem: { facet, term },
-    action
-  }) => {
-    if (action === "add") {
-      if (Object.keys(filters).includes(facet)) {
-        setFilters({
-          ...filters,
-          [facet]: { ...filters[facet], [term.term]: term }
-        });
-      } else {
-        setFilters({
-          ...filters,
-          [facet]: { [term.term]: term }
-        });
-      }
-    }
+  const addToFilter = useCallback(
+    (payload: FilterPayloadType) => {
+      setQueryParametersInUrl({
```

Comment:
Think it would be less processing if we add a check wether the "withFilters" is already there